sqlmesh/core/plan/builder.py

```diff
@@ -94,6 +103,7 @@ def __init__(
         self._skip_backfill = skip_backfill
         self._is_dev = is_dev
         self._forward_only = forward_only
+        self._allow_destructive_models = set(allow_destructive_models or [])
```

Comment:
this code is not correct given that allow_destructive_models is an iterable, iterables are never falsey. so you need to check if it's None instead

Reply:
Thank you - should I go ahead and change it for `self._restate_models` below as well?

Reply:
yes